### PR TITLE
#1509 - Followup to Notifications System PR

### LINKED ISSUE #1509: Followup to Notifications System PR
### Problem

As mentioned in [the review](https://github.com/okTurtles/group-income/pull/1497#pullrequestreview-1324904235), there are 2 remaining things to be done for the notifications system:

1. [Handling expiring proposals notification](https://github.com/okTurtles/group-income/pull/1497#discussion_r1121173815)
2. Using `dateToPeriodStamp` where appropriate (e.g. [here](https://github.com/okTurtles/group-income/pull/1497#pullrequestreview-1324866866) and [here](https://github.com/okTurtles/group-income/pull/1497#pullrequestreview-1324902737), and anywhere else I missed)

### Solution

Do those things :)

### PR DESCRIPTION
**Pull request by SebinSong**
closes #1509

### CODE REVIEW

File: frontend/model/notifications/mainNotificationsMixin.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ const periodicNotificationEntries = [
         if (!currentPeriod) { return false }
 
         const nextPeriod = rootGetters.periodAfterPeriod(currentPeriod)
-        const now = new Date().toISOString()
+        const now = dateToPeriodStamp(new Date())
```
Comment (taoeffect) [inline frontend/model/notifications/mainNotificationsMixin.js:80]: Below here on line 119:

```js
return comparePeriodStamps(new Date().toISOString(), 
```

Also needs `dateToPeriodStamp`

File: frontend/model/notifications/mainNotificationsMixin.js [hunk 2]
```diff
@@ -125,38 +125,33 @@ const periodicNotificationEntries = [
...
-              expiringProposals.push({
-                proposalId,
-                proposalType: proposal.data.proposalType,
-                proposalData: proposal.data.proposalData,
-                expires_date_ms: proposal.data.expires_date_ms,
-                createdDate: proposal.meta.createdDate,
-                creator: proposal.meta.username
-              })
-            }
+        const groupProposals = rootGetters.currentGroupState.proposals || {}
```
Comment (taoeffect) [inline frontend/model/notifications/mainNotificationsMixin.js]: Hmm.. why was this modified? Doesn't it make sense for us to emit notifications for other groups we're signed into at the moment that have expiring proposals?

File: frontend/model/notifications/mainNotificationsMixin.js [hunk 3]
```diff
@@ -125,38 +125,33 @@ const periodicNotificationEntries = [
...
-      async emit () {
-        for (const { contractID, proposals } of this.expiringProposalsByGroup) {
-          await sbp('gi.actions/group/notifyExpiringProposals', {
-            contractID,
-            data: { proposals }
-          })
-        }
+      async emit ({ rootState }) {
+        await sbp('gi.actions/group/notifyExpiringProposals', {
+          contractID: rootState.currentGroupId,
```
Comment (taoeffect) [inline frontend/model/notifications/mainNotificationsMixin.js]: I think the `contractID` here needs to be as it was before - based on the group that the notification is for

**Review by taoeffect — CHANGES_REQUESTED** [review]
Thanks @SebinSong! Left a few comments!

File: frontend/model/notifications/mainNotificationsMixin.js (hunk 2 above)
Comment (SebinSong) [reply frontend/model/notifications/mainNotificationsMixin.js]: @taoeffect It's due to the way that `gi.notifications/emit` works.
it sends the notification for the group that the user's has logged in on at the time but has no capability to send it to the other groups that are not active. (reference [here](https://github.com/okTurtles/group-income/blob/4675aab76b63524c05483e0ab1d890aa5105f24a/frontend/model/notifications/selectors.js#L15))

Comment (taoeffect) [reply frontend/model/notifications/mainNotificationsMixin.js]: Why do you say it "has no capability to send it to the other groups that are not active"?

I looked at your link, it doesn't suggest that it can't support notifications for other groups. That's why the `NotificationData` object takes a `groupID` parameter

File: frontend/model/notifications/mainNotificationsMixin.js (hunk 3 above)
Comment (SebinSong) [reply frontend/model/notifications/mainNotificationsMixin.js]: done!

File: frontend/model/notifications/mainNotificationsMixin.js (hunk 2 above)
Comment (SebinSong) [reply frontend/model/notifications/mainNotificationsMixin.js]: You are right. made additional changes accordingly.

**Review by taoeffect — APPROVED** [review]
Beautifully done @SebinSong!
